src/app/modules/calendar/calendar.model.ts

```diff
@@ -0,0 +1,19 @@
+export class CalendarModel {
```

Comment:
If you extend the AngularDesignSystemModel class, then you can make the constructor a lot simpler.  Check out the popovermodel class